src/Aspire.Dashboard/Utils/MimeTypeHelpers.cs

```diff
@@ -0,0 +1,161 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Dashboard.Utils;
+
+public static class MimeTypeHelpers
+{
+    // Browsers support a limited set of image mime types.
+    // See: https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Image_types
+    // We only allow these known safe types to avoid broken image in the browser.
+    public static readonly HashSet<string> SupportedImageTypes = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+    {
+        "image/apng",
+        "image/avif",
+        "image/bmp",
+        "image/gif",
+        "image/jpeg",
+        "image/jpg",
+        "image/png",
+        "image/svg+xml",
+        "image/webp",
+        "image/x-icon"
+    };
+
+    public static readonly HashSet<string> SupportedAudioTypes = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+    {
+        "audio/mpeg",
+        "audio/wav",
+        "audio/x-wav",
+        "audio/mp4",
+        "audio/aac",
+        "audio/ogg",
+        "audio/flac",
+        "audio/webm"
+    };
+
+    // AI generated list of extensions.
+    // Combines 100 commonly used mime types with https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types.
+    public static readonly Dictionary<string, string> MimeToExtension = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
```

Comment:
Yet another example where https://github.com/dotnet/runtime/issues/121017 will be useful.